.old..........pc/linux-2.6.3-nfs-intent-suse/fs/nfs/dir.c
.new.........fs/nfs/dir.c
.old..........pc/linux-2.6.3-nfs-intent-suse/fs/nfs/dir.c
.new.........fs/nfs/dir.c
Index: linux-2.6.3-20/fs/nfs/dir.c
===================================================================
--- linux-2.6.3-20.orig/fs/nfs/dir.c	2004-03-08 14:23:40.000000000 -0800
+++ linux-2.6.3-20/fs/nfs/dir.c	2004-03-08 17:07:34.000000000 -0800
@@ -751,7 +751,7 @@
 	if (nd->flags & LOOKUP_DIRECTORY)
 		return 0;
 	/* Are we trying to write to a read only partition? */
-	if (IS_RDONLY(dir) && (nd->intent.open.flags & (O_CREAT|O_TRUNC|FMODE_WRITE)))
+	if (IS_RDONLY(dir) && (nd->intent.it_flags & (O_CREAT|O_TRUNC|FMODE_WRITE)))
 		return 0;
 	return 1;
 }
@@ -772,7 +772,7 @@
 	dentry->d_op = NFS_PROTO(dir)->dentry_ops;
 
 	/* Let vfs_create() deal with O_EXCL */
-	if (nd->intent.open.flags & O_EXCL)
+	if (nd->intent.it_flags & O_EXCL)
 		goto no_entry;
 
 	/* Open the file on the server */
@@ -788,7 +788,7 @@
 				break;
 			/* This turned out not to be a regular file */
 			case -ELOOP:
-				if (!(nd->intent.open.flags & O_NOFOLLOW))
+				if (!(nd->intent.it_flags & O_NOFOLLOW))
 					goto no_open;
 			/* case -EISDIR: */
 			/* case -EINVAL: */
@@ -818,7 +818,7 @@
 	parent = dget_parent(dentry);
 	if (!is_atomic_open(parent->d_inode, nd))
 		goto no_open;
-	openflags = nd->intent.open.flags;
+	openflags = nd->intent.it_flags;
 	if (openflags & O_CREAT) {
 		/* If this is a negative dentry, just drop it */
 		if (!inode)
.old..........pc/linux-2.6.3-nfs-intent-suse/fs/nfs/nfs4proc.c
.new.........fs/nfs/nfs4proc.c
Index: linux-2.6.3-20/fs/nfs/nfs4proc.c
===================================================================
--- linux-2.6.3-20.orig/fs/nfs/nfs4proc.c	2004-03-05 02:07:03.000000000 -0800
+++ linux-2.6.3-20/fs/nfs/nfs4proc.c	2004-03-08 17:07:34.000000000 -0800
@@ -778,17 +778,17 @@
 	struct nfs4_state *state;
 
 	if (nd->flags & LOOKUP_CREATE) {
-		attr.ia_mode = nd->intent.open.create_mode;
+		attr.ia_mode = nd->intent.it_create_mode;
 		attr.ia_valid = ATTR_MODE;
 		if (!IS_POSIXACL(dir))
 			attr.ia_mode &= ~current->fs->umask;
 	} else {
 		attr.ia_valid = 0;
-		BUG_ON(nd->intent.open.flags & O_CREAT);
+		BUG_ON(nd->intent.it_flags & O_CREAT);
 	}
 
 	cred = rpcauth_lookupcred(NFS_SERVER(dir)->client->cl_auth, 0);
-	state = nfs4_do_open(dir, &dentry->d_name, nd->intent.open.flags, &attr, cred);
+	state = nfs4_do_open(dir, &dentry->d_name, nd->intent.it_flags, &attr, cred);
 	put_rpccred(cred);
 	if (IS_ERR(state))
 		return (struct inode *)state;
.old..........pc/linux-2.6.3-nfs-intent-suse/fs/cifs/dir.c
.new.........fs/cifs/dir.c
Index: linux-2.6.3-20/fs/cifs/dir.c
===================================================================
--- linux-2.6.3-20.orig/fs/cifs/dir.c	2004-03-05 02:07:03.000000000 -0800
+++ linux-2.6.3-20/fs/cifs/dir.c	2004-03-08 17:16:19.000000000 -0800
@@ -146,18 +146,18 @@
 	if(nd) { 
 		cFYI(1,("In create for inode %p dentry->inode %p nd flags = 0x%x for %s",inode, direntry->d_inode, nd->flags,full_path));
 
-		if ((nd->intent.open.flags & O_ACCMODE) == O_RDONLY)
+		if ((nd->intent.it_flags & O_ACCMODE) == O_RDONLY)
                 	desiredAccess = GENERIC_READ;
-		else if ((nd->intent.open.flags & O_ACCMODE) == O_WRONLY)
+		else if ((nd->intent.it_flags & O_ACCMODE) == O_WRONLY)
 			desiredAccess = GENERIC_WRITE;
-		else if ((nd->intent.open.flags & O_ACCMODE) == O_RDWR)
+		else if ((nd->intent.it_flags & O_ACCMODE) == O_RDWR)
 			desiredAccess = GENERIC_ALL;
 
-		if((nd->intent.open.flags & (O_CREAT | O_EXCL)) == (O_CREAT | O_EXCL))
+		if((nd->intent.it_flags & (O_CREAT | O_EXCL)) == (O_CREAT | O_EXCL))
 			disposition = FILE_CREATE;
-		else if((nd->intent.open.flags & (O_CREAT | O_TRUNC)) == (O_CREAT | O_TRUNC))
+		else if((nd->intent.it_flags & (O_CREAT | O_TRUNC)) == (O_CREAT | O_TRUNC))
 			disposition = FILE_OVERWRITE_IF;
-		else if((nd->intent.open.flags & O_CREAT) == O_CREAT)
+		else if((nd->intent.it_flags & O_CREAT) == O_CREAT)
 			disposition = FILE_OPEN_IF;
 		else {
 			cFYI(1,("Create flag not set in create function"));
@@ -314,7 +314,7 @@
 	      parent_dir_inode, direntry->d_name.name, direntry));
 
 	if(nd) {  /* BB removeme */
-		cFYI(1,("In lookup nd flags 0x%x open intent flags 0x%x",nd->flags,nd->intent.open.flags));
+		cFYI(1,("In lookup nd flags 0x%x open intent flags 0x%x",nd->flags,nd->intent.it_flags));
 	} /* BB removeme BB */
 	/* BB Add check of incoming data - e.g. frame not longer than maximum SMB - let server check the namelen BB */
 
